cmd/kubelet/kubelet.go

```diff
@@ -34,13 +34,21 @@ import (
 )
 
 func main() {
+	if err := runKubeletCmd(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func runKubeletCmd() error {
 	rand.Seed(time.Now().UnixNano())
 
 	command := app.NewKubeletCommand()
 	logs.InitLogs()
 	defer logs.FlushLogs()
 
 	if err := command.Execute(); err != nil {
-		os.Exit(1)
+		return err
 	}
+
+	return nil
```

Comment:
https://github.com/kubernetes/kubernetes/pull/105076/files has a similar fix that makes `main` a bit more consistent with the way how other commands are set up (command constructed in main, executed in helper function).

It's a minor difference, but the two PRs will conflict.

Can we perhaps merge the other PR first (it touches many different areas and will be harder to get merged when rebasing is needed)?

Reply:
of course, waiting for good news.

Reply:
/cancel hold

The other PR was merged, please rebase. Thanks for your patience.

Reply:
/hold cancel

Reply:
thx